build.gradle

```diff
@@ -107,24 +127,44 @@ List<String> resolveIntelliJVersions(String versionKey) {
   return [versionValue]
 }
 
-class ResolveIntelliJVersions extends DefaultTask {
-  @Input
-  @Option(option = 'version-key', description = 'Either release number (like 2020.3) or key of intellijVersions (like eapOfLatestSupportedMajor)')
-  String versionKey
-
-  @OptionValues('version-key')
-  List<String> getAvailableVersionKeys() {
-    return new ArrayList<>(project.intellijVersions.keySet())
+import java.util.regex.Matcher
+import java.util.regex.Pattern
+import org.jsoup.Jsoup
+import org.jsoup.nodes.Element
+import org.jsoup.nodes.Document
+
+String checkForNewerEap(String intellijMajorVersion) {
+  Document htmlContent = Jsoup.connect(project.intellijSnapshotsUrl).get()
+  Element link = htmlContent.select('a[href^='+project.intellijSnapshotsUrl+'com/jetbrains/intellij/idea/ideaIC/][href$="-EAP-SNAPSHOT.pom"]')[0]
+  if (link){
+    String attr = link.attr("href")
+    Pattern pattern = Pattern.compile("ideaIC-(.*?).pom")
+    Matcher matcher = pattern.matcher(attr)
+    matcher.find()
+    String foundVersion = matcher.group(1)
+    if (!intellijMajorVersion){
+      if (Integer.parseInt(foundVersion.substring(0,3)) > Integer.parseInt(intellijVersionToBuildNumber(project.intellijVersions.latestStable))) {
+        return foundVersion
+      }
+    } else {
+      return foundVersion
+    }
   }
+  return null
+}
 
-  @TaskAction
-  void print() {
-    println(String.join("\n", project.resolveIntelliJVersions(versionKey)))
+task updateEapOfLatestSupportedMajor {
+  Properties properties = intellijVersionsProperties()
+  File propFile = intellijVersionsFile()
+  String latestEap = project.intellijVersions.eapOfLatestSupportedMajor
+  String newerVersion = latestEap ? project.checkForNewerEap(latestEap.split("\\.")[0]) :
```

Comment:
And for consistency, IIUC `majorPartOfBuildNumber` can be used here as well

Reply:
hmm but when latestEap exists we need to pass whole number e.q. 221.5080.93 to checkForNewerEap method not only 221 (if i'm right?) we need to pass only 221 (from latest stable) if latestEap not exist

Reply:
Yes, indeed, the comment https://github.com/VirtusLab/git-machete-intellij-plugin/pull/841#discussion_r875993057 somewhat invalidates this one ;)